src/overlays/actors/ovl_En_Kakasi/z_en_kakasi.c

```diff
@@ -21,27 +43,301 @@ const ActorInit En_Kakasi_InitVars = {
+        case 1:
+            this->unk_19A++;
+            this->unk_1B8 = 1.0f;
+            break;
+        case 2:
+            this->unk_19A++;
+            if (this->unk_1AC == 0x0) {
+                this->unk_1AC = 0x1388;
+            }
+            break;
+        case 3:
+            this->unk_19A++;
+            if (this->unk_1A8 == 0x0) {
+                this->unk_1A8 = 0x1388;
+            }
+            break;
+        case 4:
+            this->unk_19A++;
+            this->unk_1B8 = 2.0f;
+            break;
+    }
+
+    if (this->unk_19A > 8) {
+        this->unk_19A = 8;
+    }
+
+    if (this->unk_19A != 0) {
+        this->actor.gravity = -1.0f;
+        if (this->unk_19A == 8 && (this->actor.bgCheckFlags & 1)) {
+            this->actor.velocity.y = 3.0f;
+            Audio_PlayActorSound2(&this->actor, NA_SE_IT_KAKASHI_JUMP);
+        }
+        Math_SmoothScaleMaxF(&this->skelanime.animPlaybackSpeed, this->unk_1B8, 0.1f, 0.2f);
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.x, this->unk_1A8, 0x5, 0x3E8, 0);
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.z, this->unk_1AC, 0x5, 0x3E8, 0);
+
+        if (this->unk_1A8 != 0x0 && fabsf(this->actor.shape.rot.x - this->unk_1A8) < 50.0f) {
+            this->unk_1A8 *= -1.0f;
+        }
+        if (this->unk_1AC != 0x0 && fabsf(this->actor.shape.rot.z - this->unk_1AC) < 50.0f) {
+            this->unk_1AC *= -1.0f;
+        }
+
+        if (this->unk_1A4 != 0) {
+            this->actor.shape.rot.y += 0x1000;
+            if (this->actor.shape.rot.y == 0) {
+                this->unk_1A4 = 0;
+            }
+        }
+        currentFrame = this->skelanime.animCurrentFrame;
+        if (currentFrame == 11 || currentFrame == 17) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EV_KAKASHI_SWING);
+        }
+        SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    }
+}
+
+void func_80A8F660(EnKakasi* this, GlobalContext* globalCtx) {
+    f32 frameCount = SkelAnime_GetFrameCount(&D_06000214.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelanime, &D_06000214, 1.0f, 0.0f, (s16)frameCount, 0, -10.0f);
+
+    this->actor.textId = 0x4076;
+    this->unk_196 = 6;
+    if (LINK_IS_CHILD) {
+        this->unk_194 = false;
+        if (gSaveContext.unk_F3C[4] != 0) {
+            this->actor.textId = 0x407A;
+            this->unk_196 = 5;
+        }
+    } else {
+        this->unk_194 = true;
+        if (gSaveContext.unk_F3C[4] != 0) {
+            this->actor.textId = 0x4079;
+            this->unk_196 = 5;
+        }
+    }
+    this->actionFunc = func_80A8F75C;
+}
+
+void func_80A8F75C(EnKakasi* this, GlobalContext* globalCtx) {
+    Player* player = PLAYER;
+    s16 angleTowardsLink;
+    s16 absAngleTowardsLink;
+
+    func_80A8F28C(this);
+    SkelAnime_FrameUpdateMatrix(&this->skelanime);
+    this->camId = -1;
+    if (func_8002F194(&this->actor, globalCtx)) {
+        if (this->unk_196 == 5) {
+            this->actionFunc = &func_80A8F9C8;
```

Comment:
```suggestion
            this->actionFunc = func_80A8F9C8;
```
no need for `&` for function pointers

also in  a few other places in the file